tests/regression_tests/unstructured_mesh/source_sampling/test.py

```diff
@@ -0,0 +1,114 @@
+from itertools import product
+
+from pathlib import Path
+import pytest
+import numpy as np
+
+import openmc
+import openmc.lib
+
+from tests import cdtemp
+from tests.testing_harness import PyAPITestHarness
+
+TETS_PER_VOXEL = 12
+
+# This test uses a geometry file with cells that match a regular mesh. Each cell
+# in the geometry corresponds to 12 tetrahedra in the unstructured mesh file.
+@pytest.fixture
+def model():
+    openmc.reset_auto_ids()
+
+    ### Materials ###
+    materials = openmc.Materials()
+
+    water_mat = openmc.Material(material_id=3, name="water")
+    water_mat.add_nuclide("H1", 2.0)
+    water_mat.add_nuclide("O16", 1.0)
+    water_mat.set_density("atom/b-cm", 0.07416)
+    materials.append(water_mat)
+
+    ### Geometry ###
+    # create a regular mesh that matches the superimposed mesh
+    regular_mesh = openmc.RegularMesh(mesh_id=10)
+    regular_mesh.lower_left = (-10, -10, -10)
+    regular_mesh.dimension = (10, 10, 10)

```

Comment:
This results in 1000 elements/cells -- maybe let's make this a bit smaller

Reply:
That's a little tricky to do unless we change the unstructured meshes associated with these tests (i.e. `test_mesh_tets.e`). I could add a smaller mesh, but we already have these in the repo.

Reply:
Ah yeah, I see. Ok, no worries!